homeassistant/components/knx/switch.py

```diff
@@ -54,6 +55,15 @@ def __init__(self, xknx: XKNX, config: ConfigType) -> None:
         )
         self._unique_id = f"{self._device.switch.group_address}"
 
+    async def async_added_to_hass(self) -> None:
+        """Restore last state."""
+        await super().async_added_to_hass()
+        if not self._device.switch.readable and (
+            last_state := await self.async_get_last_state()
+        ):
+            if last_state.state not in (STATE_UNKNOWN, STATE_UNAVAILABLE):
+                self._device.switch.value = last_state.state == STATE_ON
```

Comment:
If that is the case, shouldn't this be an assumed state entity when the device isn't readable?

Reply:
We have basically 2 kinds of switches without `state_address`:
- Representing binary values from older actuators that don't support reading properly. The state of such a switch is unknown until the first telegram on a configured group address is received (or sent). Then it is known until a disconnect. Since we [should not change `assumed_state` at runtime](https://github.com/home-assistant/core/pull/43536#discussion_r528406212) it wouldn't fit here.
- A switch providing state itself. In combination with #51790 we can create a KNX switch that provides the single source of truth to the bus out of HA. KNX devices can read and act on this state. E.g. enable/disable heating, or just providing some input for KNX logic servers. Basically every KNX visualization provides such a feature to create a "virtual switch".
I'm not sure if `assumed_state` is a good fit here either. I like the UI representation of a non-assumed toggle better for these things since its state is more obvious - and this is what can be read over KNX.